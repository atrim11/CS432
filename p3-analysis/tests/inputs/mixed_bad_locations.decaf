int arr[300];
def int add(int a, int b) {
    return a + b;
}

def int main() {
    // array access no idx
    int a;
    void f;
    int d;
    a = arr;
    // local var accessed as array
    d = a[1];
    // nonfunction call
    a = f(69);
    // non-array access with idx
    a[300] = 5;
    // func accessed as index
    add[300] = 5;
    a = add[300];
    // func accessed as variable
    f = add;
    d = add;
    f = f + d;
    return f;
}