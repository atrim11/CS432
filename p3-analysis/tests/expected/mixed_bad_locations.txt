Void variable 'f' on line 9
Array 'arr' accessed without index on line 11
Non-array 'a' accessed as an array on line 13
Invalid call to non-function 'f' on line 15
Invalid number of function arguments on line 15
Type mismatch: int is incompatible with void on line 15
Non-array 'a' accessed as an array on line 17
Non-array 'add' accessed as an array on line 19
Non-array 'add' accessed as an array on line 20
Function 'add' accessed as a variable on line 22
Type mismatch: void is incompatible with int on line 22
Function 'add' accessed as a variable on line 23
Type mismatch: int expected but void found on line 24
Type mismatch: void is incompatible with int on line 24
Type mismatch: int expected but void found on line 25
